packages/core/src/browser/dialogs/react-dialog.tsx

```diff
@@ -0,0 +1,58 @@
+/********************************************************************************
+ * Copyright (C) 2019 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as ReactDOM from 'react-dom';
+import * as React from 'react';
+import { injectable, inject } from 'inversify';
+import { DisposableCollection, Disposable } from '../../common';
+import { Message } from '../widgets';
+import { AbstractDialog } from './abstract-dialog';
+import { DialogProps } from './dialogs';
+
+// Modified version of ReactWidget, based on AbstractDialog
+
+@injectable()
+export abstract class ReactDialog extends AbstractDialog<void> {
+    protected readonly onRender = new DisposableCollection();
+
+    constructor(
+        @inject(DialogProps) protected readonly props: DialogProps
+    ) {
+        super(props);
+        this.scrollOptions = {
+            suppressScrollX: true,
+            minScrollbarLength: 35,
+        };
+        this.toDispose.push(Disposable.create(() => {
+            ReactDOM.unmountComponentAtNode(this.contentNode);
+        }));
+    }
+
+    protected onUpdateRequest(msg: Message): void {
+        super.onUpdateRequest(msg);
+        ReactDOM.render(<React.Fragment>{this.render()}</React.Fragment>, this.contentNode, () => this.onRender.dispose());
+    }
+
+    get value(): undefined { return undefined; }
+
```

Comment:
Would it be better to parameterize the value?  So we have something like:

```ts
export interface ReactDialogProps<V> extends DialogProps {
    initialValue: V;
}

export abstract class ReactDialog<V> extends AbstractDialog<Readonly<V>> {
    protected _value: V;


    protected constructor(
        @inject(DialogProps) protected readonly props: DialogProps<V>
    ) {
        super(props);
        this._value = props.initialValue;
    }

...
    protected onValueChange(newValue: V): void {
        this._value = newValue;
        this.update();
    }

    get value(): V {
        return this._value;
    }
}```

What do you think?

Reply:
Why do we need `value` here at all? It is not used. I don't think we should add unused code, subclasses can add necessary properties if necessary.